src/App.tsx

```diff
@@ -1,24 +1,157 @@
-import React from 'react';
-import logo from './logo.svg';
-import './App.css';
+import React, { useEffect, useState } from "react";
+import "antd/dist/antd.min.css";
+import "./App.css";
+import { Recipe } from "./recipe";
+import { Button, Collapse, Image } from "antd";
+import { DeleteOutlined, EditOutlined } from "@ant-design/icons";
 
 function App() {
+  const url = `${process.env.REACT_APP_URL}`;
+  if (url === undefined) {
+    throw new Error("invalid URL");
+  }
+  const [allRecipes, setAllRecipes] = useState<Recipe[]>([]);
+
+  const getAllRecipes = async () => {
+    const response = await fetch(url);
+    const data: Recipe[] = await response.json();
+    if (data === undefined) {
+      throw new Error("bad api call");
+    } else {
```

Comment:
I don't think the else is necessary here, since there isn't code executed after it